lib/std/mem.zig

```diff
@@ -876,7 +870,86 @@ pub fn indexOfPos(comptime T: type, haystack: []const T, start_index: usize, nee
     return null;
 }
 
+fn boyerMooreHorspoolPreprocessReverse(pattern: []const u8, table: *[256]usize) void {
+    for (table) |*c| {
+        c.* = pattern.len;
+    }
+
+    var i: usize = pattern.len - 1;
+    while (i > 0) : (i -= 1) {
```

Comment:
This look doesn't process pattern[0].  Is that intentional?  If so, you should leave a comment.

Reply:
Yes its intentional, its the standard way its implemented where the first/last element is pattern.len. Added comment to explain that now